kuma/javascript/src/payments/subscription-form.jsx

```diff
@@ -1,53 +1,123 @@
 // @flow
 import * as React from 'react';
-import { useRef, useState } from 'react';
+import { useContext, useEffect, useRef, useState } from 'react';
 
 import { getLocale, gettext, Interpolated } from '../l10n.js';
+import UserProvider from '../user-provider.jsx';
+import { getCookie } from '../utils';
+
+const SUBSCRIPTION_URL = '/api/v1/subscriptions';
+
+/**
+ * Conditionally loads the script given by the URL and cleans up after itself
+ * @returns {(Promise|null)} When it shouldLoad a promise indicating whether
+ * the script has successfully loaded
+ */
+function useConditionalScriptLoad(url, shouldLoad) {
+    const [loadingPromise, setLoadingPromise] = useState(null);
+
+    useEffect(() => {
+        let script;
+        if (shouldLoad && !loadingPromise) {
+            setLoadingPromise(
+                new Promise(resolve => {
+                    script = document.createElement('script');
+                    script.onload = () => {
+                        resolve();
+                    };
+                    script.src = url;
+
+                    if (document.head) {
+                        document.head.appendChild(script);
+                    }
+                })
+            );
+        }
+        return () => {
+            if (document.head && script) {
+                document.head.removeChild(script);
+            }
+        };
+    }, [shouldLoad, loadingPromise, url]);
+
+    return loadingPromise;
+}
 
 export default function SubscriptionForm() {
     const locale = getLocale();
-    const subscriptionFormRef = useRef(null);
+    const userData = useContext(UserProvider.context);
+
     const [isSubmitting, setIsSubmitting] = useState(false);
     const [paymentAuthorized, setPaymentAuthorized] = useState(false);
 
-    const toggleButton = () => {
-        setPaymentAuthorized(!paymentAuthorized);
-    };
+    const token = useRef(null);
 
-    /**
-     * Opens Stripe modal allowing a user to complete their subscription.
-     * @param {Object} event - The form submit event
-     */
-    const submit = event => {
+    const stripeLoadingPromise = useConditionalScriptLoad(
+        'https://checkout.stripe.com/checkout.js',
+        paymentAuthorized
+    );
+
+    function togglePaymentAuthorized() {
+        const newValue = !paymentAuthorized;
+        setPaymentAuthorized(newValue);
+    }
+
+    function openStripeModal(event) {
         event.preventDefault();
 
-        const subscriptionForm = subscriptionFormRef.current;
-        if (subscriptionForm) {
-            const formData = new FormData(subscriptionForm);
-            setIsSubmitting(true);
+        setIsSubmitting(true);
+
+        if (!stripeLoadingPromise) {
+            console.error('stripe script load was not started');
```

Comment:
My limited mind is still struggling to understand how this can happen. 
And even if it could, I don't think we should just swallow like that without any UI feedback.

Reply:
Yeah unfortunately no type system way of enforcing it came to mind. Good point, I'll change it to display something :+1: